raiden-ts/src/services/utils.ts

```diff
@@ -103,42 +91,64 @@ export function pfsInfo(
-      // default to https for domain-only urls
-      else if (!url.startsWith('https://') && !url.startsWith('http://')) url = `https://${url}`;
+  return url$
+    .pipe(
+      withLatestFrom(config$),
+      mergeMap(([url, { httpTimeout, pfsMaxFee }]) => {
+        if (!url) throw new RaidenError(ErrorCodes.PFS_EMPTY_URL);
+        else if (!isValidUrl(url)) throw new RaidenError(ErrorCodes.PFS_INVALID_URL, { url });
+        // default to https for domain-only urls
+        else if (!url.startsWith('https://') && !url.startsWith('http://')) url = `https://${url}`;
 
-      const start = Date.now();
-      return fromFetch(url + '/api/v1/info').pipe(
-        timeout(httpTimeout),
-        mergeMap(
-          async (res) =>
-            [
-              decode(PathInfo, jsonParse(await res.text())),
-              await serviceRegistryToken(serviceRegistryContract, provider.pollingInterval),
-            ] as const,
-        ),
-        map(([info, token]) => {
-          assert(info.price_info.lte(pfsMaxFee), [
-            ErrorCodes.PFS_TOO_EXPENSIVE,
-            { price: info.price_info.toString() },
-          ]);
-          return {
-            address: info.payment_address,
-            url,
-            rtt: Date.now() - start,
-            price: info.price_info,
-            token,
-          };
-        }),
-      );
-    }),
-  );
+        const start = Date.now();
+        return fromFetch(url + '/api/v1/info').pipe(
+          timeout(httpTimeout),
+          mergeMap(
+            async (res) =>
+              [
+                decode(PathInfo, jsonParse(await res.text())),
+                await serviceRegistryToken(serviceRegistryContract, provider.pollingInterval),
+              ] as const,
+          ),
+          map(([info, token]) => {
+            assert(info.price_info.lte(pfsMaxFee), [
+              ErrorCodes.PFS_TOO_EXPENSIVE,
+              { price: info.price_info.toString() },
+            ]);
+            return {
+              address: info.payment_address,
+              url,
+              rtt: Date.now() - start,
+              price: info.price_info,
+              token,
+            };
+          }),
+        );
+      }),
+    )
+    .toPromise();
 }
 
+/**
+ * Returns a cold observable which fetch PFS info & validate for a given server address or URL
+ *
+ * This is a memoized function which caches by url or address, network and registry used.
+ *
+ * @param pfsAddrOrUrl - PFS account/address or URL
+ * @param deps - RaidenEpicDeps needed for various parameters
+ * @param deps.serviceRegistryContract - ServiceRegistry contract instance
+ * @param deps.network - Current Network
+ * @param deps.contractsInfo - ContractsInfo mapping
+ * @param deps.config$ - Config observable
+ * @param deps.provider - Eth provider
+ * @returns Observable containing PFS server info
+ */
+export const pfsInfo: typeof pfsInfo_ = memoize(
```

Comment:
The info object for a PFS does contain some variable data like. I'm lacking knowledge, but how would we get updated about such properties like the prize?

Reply:
Currently, it isn't, at least on a single session, we just assume PFS isn't changing that often, but we could have some ttl cache.

Reply:
So this actually means the user could confirm on the dApp to use a PFS with a prize that is not up to date anymore?

Reply:
It could be higher than you wanted, but you would have reviewed and agreed with that higher value (either interactively on dApp or on CLI through `--pathfindingMaxFee` option), so I'd consider this a race which is not common and reasonable to be accepted. In case it's too small, it may fail until the cache expires, but next PFS should be tried, I need to check this though or implement retrying with clean cache in case of this specific error, which can also be considered a race (although a up to 1h long one, currently).

Reply:
Alternatively, we could implement a dedicated function to request this endpoint (instead of reusing `pfsInfo`) which would cache only `address`, which is for what we care about this (for sending messages through toDevice on `next`).

Reply:
Maybe we could discuss this in person again? Your last comment sounds quite interesting. Else I would say this a too big and critical feature addition which deserves its own PR and should not just be added without a note.

Reply:
I've implemented it. Again, had to rebase due to the required strict order of the `next:` commit. `pfsInfo` isn't cached anymore, but sets the cache for `address`. `pfsInfoAddress` is the provided cached util, which uses `pfsInfo` and extracts only the `address` (which doesn't change) but doesn't prevent any following `pfsInfo` call to perform a new request. This should take the best of both cases.

Reply:
Sounds reasonable for me. Good job. 👍🏾